connection.go

```diff
@@ -153,7 +153,7 @@ type connection struct {
 	unpacker      unpacker
 	frameParser   wire.FrameParser
 	packer        packer
-	mtuDiscoverer mtuDiscoverer // initialized when the handshake completes
+	mtuDiscoverer mtuDiscoverer // initialized when the transport parameters are received
```

Comment:
Note that the original comment here was incorrect: We used to initialize the `mtuDiscoverer` with the connection, not on handshake completion.